[carte-v0.5 branch] Add map component for map section

JesseEmond opened the pull request:
- Adjust styling of map zoom controls to match design (and fit with component next to it);
- Remove map scale to match design;
- Add an initial (static) catastrophe toggle header. Will later make it open the pop-up and give it real values;
- Add back the RegionSearch, with adjusted styling, now on top of map;
- Remove thermometer from map for now -- will get its own section.
- Currently add margin on region search to allow space for attribution -- should be explicitly incorporated in the design instead.

Visually:
![image](https://user-images.githubusercontent.com/1843555/189513791-776fe844-7401-4a9e-a925-af493662fe4d.png)

Commit:
Add map component for map section

Adjust styling of map controls to match design (and fit with component
next to it).
Add an initial (static) catastrophe toggle header.
Add back the RegionSearch, with adjusted styling.
Remove thermometer from map for now.

edemartel commented:
L'attribution est obligatoire à cause du contrat d'utilisation d'OpenStreetMap, faque va falloir trouver une façon de fitter ça dans le design.

Commit:
Restore attribution & fix map interactions

Attribution is required.
With the map overlay as-is, the map can't be interacted with anymore.
Disable pointer events on the overall overlay and retore them on
relevant elements.

JesseEmond commented:
> L'attribution est obligatoire à cause du contrat d'utilisation d'OpenStreetMap, faque va falloir trouver une façon de fitter ça dans le design.

Oui merci beaucoup d'avoir spotté ça! Restored, avec une margin pour y faire de la place pour l'instant.
